ui/src/components/SampleGrid/SampleCircleView.jsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect } from 'react';
+import { Col } from 'react-bootstrap';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { filterAction } from '../../actions/sampleGrid';
+
+export default function SampleCircleView(props) {
+  const { displayPuckCellContextMenu, puckMenuID, type = 'Mockup' } = props;
+
+  const filterOptions = useSelector((state) => state.sampleGrid.filterOptions);
+  const scContent = useSelector((state) => state.sampleChanger.contents);
+
+  const dispatch = useDispatch();
+
+  function handleClickOnCellPuck(event, cellID, puckID) {
+    dispatch(
+      filterAction({ cellFilter: `${cellID}`, puckFilter: `${puckID}` }),
+    );
+    event.stopPropagation();
+  }
+
+  function isPuckSelected(cellID, puckID) {
+    return (
+      Number(filterOptions.cellFilter) === cellID &&
+      Number(filterOptions.puckFilter) === puckID
+    );
+  }
+
+  useEffect(() => {
+    if (filterOptions.cellFilter === '') {
+      dispatch(filterAction({ cellFilter: `${1}`, puckFilter: `${1}` }));
+    }
+  }, [filterOptions.cellFilter, dispatch]);
+
+  function handleDisplayPuckCellContextMenu(e, menuID, cellID, puckID) {
+    e.preventDefault();
+    handleClickOnCellPuck(e, cellID, puckID === null ? '' : puckID);
+    displayPuckCellContextMenu(e, menuID, cellID, puckID);
+    e.stopPropagation();
+  }
+
+  function getSingleCellAsCircle(centerX = 10, centerY = 10) {
+    const pucks = scContent.children;
+    const totalPucks = pucks.length;
+    const cellID = 1;
+
+    // Adjust angle range based on puck count
+    const useFullCircle = totalPucks >= 8;
+    const angleRange = useFullCircle ? 2 * Math.PI : Math.PI * 1.5;
+    const startAngle = useFullCircle ? 0 : (Math.PI - angleRange) / 2;
+
+    // Circle size and spacing
+    const baseRadius = 4;
+    const placementRadius = baseRadius + Math.max(0, (totalPucks - 16) * 0.3);
+    const puckRadius = Math.min(
+      0.7,
+      (2 * Math.PI * placementRadius) / (totalPucks * 2),
+    );
+
+    // Fix angle spacing: avoid skipping puck 1 or overlapping at end
+    const angleDivisor = useFullCircle
+      ? totalPucks
+      : Math.max(1, totalPucks - 1);
+
+    return pucks.map((_, i) => {
+      const angle = startAngle + (angleRange * i) / angleDivisor;
+      const x = centerX + placementRadius * Math.cos(angle);
+      const y = centerY + placementRadius * Math.sin(angle);
+      const puckID = i + 1;
+
+      return (
+        <g
+          key={`single_cell_puck_${puckID}`}
+          onClick={(e) => handleClickOnCellPuck(e, cellID, puckID)}
+          onContextMenu={(e) =>
+            handleDisplayPuckCellContextMenu(e, puckMenuID, cellID, puckID)
+          }
+          fill={isPuckSelected(cellID, puckID) ? '#015f9d' : '#cdced1'}
```

Comment:
Might be simpler to turn this function into a variable inside the `map` callback:

```ts
const isPuckSelected = Number(filterOptions.cellFilter) === cellID && Number(filterOptions.puckFilter) === puckID;
```

Reply:
The current implementation is fine as is, no need to change it, but FYI, another way to do this that would have kept concerns separated a bit more (markup vs styles) would have been to add a `data` attribute to the `g` element and use CSS to change the fill color:

```jsx
data-selected={isPuckSelected || undefined}
```

```css
.puck {
  fill: #015f9d;
}

.puck[data-selected] {
  fill: #cdced1;
}

.puck > text {
  fill: white;
}

.puck[data-selected] > text {
  fill: black;
}
```